### DISCUSSION TITLE
Is it possible to do batch search in vss_search or vss_range_search?

### DISCUSSION THREAD

**Discussion by terry-tao-wd** [asg017/sqlite-vss#108] [Q&A] [answered]
Hi, we're using sqlite-vss to search embeddings. I'm wondering if we can pass a list of vectors to vss_search (or vss_range_search), and get back the nearest vectors of each of the passed-in vector?

I tried to pass in a list of vectors in `json` function, like this:
```
select rowid, distance
from foo
where vss_search(bar, json('[[1, 2, 3], [3, 2, 1]]'))
limit 20;
```

But got back this error:
```
Runtime error: 1st argument is not a vector
```

Is it that I'm doing it incorrectly or sqlite-vss does not support this feature right now? Thanks a lot!

**Comment by asg017** [comment, marked as answer]
hey, not currently supported. You might be able to do a `JOIN` and split your input vectors with `json_each()`, but I haven't tested it. 

Faiss has good support for searching multiple vectors at one, I'll see how hard it would be to add. Will track in https://github.com/asg017/sqlite-vss/issues/109

**Reply by terry-tao-wd** [reply]
Hi, thanks a lot for the quick response! I just figured out a way to accomplish this, by `union`-ing the queries, something like:

```
(select rowid, distance, 1 as 'qid'
from foo
where vss_search(bar, json('[1, 2, 3]'))
limit 20;)
union
(select rowid, distance, 2 as 'qid'
from foo
where vss_search(bar, json('[3, 1, 1]'))
limit 20;)
```

and then group the results by the extra 'qid' column. Although I'm not sure if there's a limit to the number of queries that I can union, or other limit, have to do some tests.

But it would definitely be good if sqlite-vss supports it natively. Thanks!
